pgjdbc/src/test/java/org/postgresql/test/jdbc2/DriverTest.java

```diff
@@ -275,6 +277,40 @@ public void testSetLogStream() throws Exception {
     }
   }
 
+  @Test
+  public void testSystemErrIsNotClosedWhenCreatedMultipleConnections() throws Exception {
+    TestUtil.initDriver();
+    PrintStream err = System.err;
+    String loggerLevel = System.getProperty("loggerLevel");
+    String loggerFile = System.getProperty("loggerFile");
+
+    System.clearProperty("loggerLevel");
+    System.clearProperty("loggerFile");
+    System.setProperty("loggerLevel", "INFO");
+    PrintStream buffer = new PrintStream(new ByteArrayOutputStream());
+    System.setErr(buffer);
+    try {
+      Connection con = DriverManager.getConnection(TestUtil.getURL());
+      try {
+        assertNotNull(con);
+      } finally {
+        con.close();
+      }
+      con = DriverManager.getConnection(TestUtil.getURL());
+      try {
+        assertNotNull(con);
+        System.err.println("");
+        assertFalse(System.err.checkError());
```

Comment:
Some helpful message could help to detect issues and make the test more clear:
`assertFalse("The System.err should not be closed", System.err.checkError());`